exotic.py

```diff
@@ -605,13 +684,50 @@ def plate_solution(fits_file, saveDirectory):
         time.sleep(5)
 
 
-# Getting the right ascension and declination for every pixel in imaging file if there is a plate solution
-def get_radec(hdulWCS):
-    wcsheader = WCS(hdulWCS[0].header)
-    xaxis = np.arange(hdulWCS[0].header['NAXIS1'])
-    yaxis = np.arange(hdulWCS[0].header['NAXIS2'])
-    x, y = np.meshgrid(xaxis, yaxis)
-    return wcsheader.all_pix2world(x, y, 0)
+# Getting the right ascension and declination for every pixel in imaging file if there is a plate solution,
+# DECOMISSIONED FOR NOW
+# def get_radec(wcsfile):
+#     hdulWCS = fits.open(wcsfile)
+#     wcsheader = WCS(hdulWCS[0].header)
+#     xaxis = np.arange(hdulWCS[0].header['NAXIS1'])
+#     yaxis = np.arange(hdulWCS[0].header['NAXIS2'])
+#     x, y = np.meshgrid(xaxis, yaxis)
+#     ra, dec = wcsheader.all_pix2world(x, y, 0)
+#     return ra, dec
+
+
+# Check the ra and dec against the plate solution to see if the user entered in the correct values
+def check_targetpixelwcs(pixx, pixy, expra, expdec, hdul):
+    while True:
+        try:
+            wcsheader = WCS(hdul[0].header)
+            obsra, obsdec = wcsheader.all_pix2world(pixx, pixy, 0)
+
+            # Margins are within 20 arcseconds ~ 0.00556 degrees
+            if expra - 0.00556 >= obsra or obsra >= expra + 0.00556:
+                print('\nError: The X Pixel Coordinate entered does not match the right ascension.')
+                raise ValueError
+            if expdec - 0.00556 >= obsdec or obsdec >= expdec + 0.00556:
+                print('\nError: The Y Pixel Coordinate entered does not match the declination.')
+                raise ValueError
+            return pixx, pixy
+        except ValueError:
+            pixx = user_input("Please re-enter the target star's X Pixel Coordinate: ", type_=int)
+            pixy = user_input("Please re-enter the target star's Y Pixel Coordinate: ", type_=int)
+
```

Comment:
Pixel coordinates will have to be fixed here, not allowing the user to re-enter the wrong coordinate. This makes the while loop only iterate twice.

Reply:
Few comments:

1. Why does the code ask the user to input the target star's x/y pixel location when it already *knows* the location of the target/comp star? Since you've done a WCS solution to the image, every pixel is registered to a RA/DEC location, so EXOTIC can exactly find by itself the location of the target/comps on the focal plane without any user input.

2. What happens if the star has moved a lot (due to proper motion) and is not actually at the pixel location that EXOTIC thinks? In other words, you should raise an error and give the user the option to proceed with their pixel coordinates or to re-enter. Right now, you only let the user re-enter new pixel values.

Reply:
1. Although EXOTIC does know where the ra/dec is for the target stars due to having all the pixels converted from another function I decommissioned, I thought when we discussed not having to automate everything, this was one of those things. I can automate it if need be to find the locations if the user entered them incorrectly. 

2. Yeah, for now it is hardstuck on having them re-enter if it doesn't deem their pixel coordinates to be in the same ra/dec. The reason why is because it goes off the 'firstimage' that it sees it can fit a centroid to. It gives the user a downloaded file of the newfits.fits to look over it and re-enter the coordinates. I can make this optional of course since I can see it running into issues.

Reply:
1. I would have EXOTIC perhaps suggest the pixel x/y coordinates that it *thinks* the target star is. That way the user can check their values against EXOTIC and have the option to either: 1. keep their original x/y position, 2. adopt EXOTIC's suggested values, or 3. enter completely new x/y pixel coords.

2. I would still add in the option for the user to use their originally-entered pixel coordinates, but point the user to the WCS-solved and aligned FITS file to double check.

Reply:
That sounds great to me!

Reply:
Corrected to do your 1. and 2.